src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -0,0 +1,279 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+from typing import List, Any, Tuple, Protocol, Union
+from snowflake.connector.options import pandas as pd
+from dateutil import parser
```

Comment:
I remember dateutil is not a required dependency of snowpark, it's from the optional pandas depdency.
can you double check it?

Reply:
it is a required package for snowpark IR, so I think it is safe it use it here